File: Shopee project/Shopee2/src/main/java/fptu/shopee2/Model/ProductPakage/Product.java
package fptu.shopee2.Model.ProductPakage;

import fptu.shopee.Model.ManyToManyRelationshipTable.CategoryValue;
import fptu.shopee.Model.ManyToManyRelationshipTable.ProductGroup;
import fptu.shopee.Model.ManyToManyRelationshipTable.ProductShippingMethod;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

import java.util.Set;

@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_product")
    private int idProduct;

    @NotNull
    @Column(name = "name_product", nullable = false)
    private String nameProduct;

    @NotNull
    @Column(name = "product_description", nullable = false)
    private String productDescription;

    @NotNull
    @Column(name = "pre_order", nullable = false)
    private Boolean preOrder = false;

    @NotNull
    @Column(name = "product_condition", nullable = false)
    private String condition;

    @Column(name = "SKU_product")
    private String skuProduct;

    @ManyToOne
    @JoinColumn(name = "id_category")
    private Category category;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<Classification> classifications;

    @OneToOne(mappedBy = "product", cascade = CascadeType.ALL)
    private ImageAndShortVideoProduct imageAndShortVideoProduct;

    @OneToMany(mappedBy = "id.product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<ProductShippingMethod> productShippingMethods;

    @OneToMany(mappedBy = "id.product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<ProductGroup> productGroups;

    @OneToMany(mappedBy = "id.product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private Set<CategoryValue> categoryValues;

    public int getIdProduct() {
        return idProduct;
    }

    public void setIdProduct(int idProduct) {
        this.idProduct = idProduct;
    }

    public String getNameProduct() {
        return nameProduct;
    }

    public void setNameProduct(String nameProduct) {
        this.nameProduct = nameProduct;
    }

    public String getProductDescription() {
        return productDescription;
    }

    public void setProductDescription(String productDescription) {
        this.productDescription = productDescription;
    }

    public Boolean getPreOrder() {
        return preOrder;
    }

    public void setPreOrder(Boolean preOrder) {
        this.preOrder = preOrder;
    }

    public String getCondition() {
        return condition;
    }

    public void setCondition(String condition) {
        this.condition = condition;
    }

    public String getSkuProduct() {
        return skuProduct;
    }

    public void setSkuProduct(String skuProduct) {
        this.skuProduct = skuProduct;
    }
}

File: .NET/Project learn/test_asp.net_Shopee_by_SQl Server/Shopee_by_SQl Server_ASP.Net/Shopee_by_SQl Server_ASP.Net/Models/Product.cs
using System;
using System.Collections.Generic;

namespace Shopee_by_SQl_Server_ASP.Net.Models;

public partial class Product
{
    public int IdProduct { get; set; }

    public string NameProduct { get; set; } = null!;

    public string ProductDescription { get; set; } = null!;

    public int? PreOrder { get; set; }

    public string ConditionProduct { get; set; } = null!;

    public string? SkuProduct { get; set; }

    public int? IdCategory { get; set; }

    public int IdShop { get; set; }

    public virtual ICollection<CategoryValue> CategoryValues { get; set; } = new List<CategoryValue>();

    public virtual ICollection<Classification> Classifications { get; set; } = new List<Classification>();

    public virtual Category? IdCategoryNavigation { get; set; }

    public virtual Shop IdShopNavigation { get; set; } = null!;

    public virtual ImageAndShortVideoProduct? ImageAndShortVideoProduct { get; set; }

    public virtual ICollection<ProductGroup> ProductGroups { get; set; } = new List<ProductGroup>();
}
